src/renderer/views/ProjectsList/subcomponents/ProjectsTable/TableHeaders.tsx

```diff
@@ -1,34 +1,106 @@
-import { TableHead, TableRow, Typography } from '@mui/material';
+import {
+  Button,
+  MenuItem,
+  TableHead,
+  TableRow,
+  Typography,
+} from '@mui/material';
 import { t } from 'i18next';
 import React from 'react';
+import { Publication } from 'src/shared/types';
 import TableCell from '../../../../components/TableCell/TableCell';
+import { SORTABLE_FIELD, SortParams } from '../../types';
+import { HeaderFilterSelect } from './style';
 
-const TableHeader = () => (
+interface Props {
+  sortParams: SortParams;
+  statusFilterValue: Publication['status'] | null;
+  setSortParams: (params: SortParams) => void;
+  setStatusFilterValue: (params: Publication['status'] | null) => void;
+}
+
+const orderSymbols: { [key in SortParams['direction']]: string } = {
+  ascending: '\xa0↑',
+  descending: '\xa0↓',
+};
+
+const TableHeader = ({
+  sortParams,
+  statusFilterValue,
+  setSortParams,
+  setStatusFilterValue,
+}: Props) => (
   <TableHead>
     <TableRow>
       <TableCell />
-      <TableCell>
-        <Typography variant='caption' component='p'>
-          {t('publication.title').toLocaleUpperCase()}
-        </Typography>
-      </TableCell>
-      <TableCell align='center'>
-        <Typography variant='caption' component='p'>
-          {t('publication.created_on').toLocaleUpperCase()}
-        </Typography>
-      </TableCell>
-      <TableCell align='center'>
-        <Typography variant='caption' component='p'>
-          {t('publication.change_date').toLocaleUpperCase()}
-        </Typography>
-      </TableCell>
-      <TableCell align='center'>
-        <Typography variant='caption' component='p'>
-          {t('publication.status').toLocaleUpperCase()}
-        </Typography>
+      {(Object.values(SORTABLE_FIELD) as SORTABLE_FIELD[]).map((field) => (
+        <TableCell
+          key={field}
+          sx={{
+            paddingLeft: '1rem',
+            textAlign: field === 'title' ? 'left' : 'center',
+          }}
+        >
+          <Button
+            variant='text'
+            sx={{
+              padding: 0,
+              minWidth: 0,
+              textAlign: field === 'title' ? 'left' : 'center',
+            }}
+            onClick={() =>
+              setSortParams({
+                field,
+                direction:
+                  sortParams.field === field
+                    ? reverseDirection(sortParams.direction)
+                    : 'ascending',
+              })
+            }
+          >
+            <Typography variant='caption' component='p'>
+              {t(`publication.${field}`)}
+              {sortParams.field === field ? (
+                orderSymbols[sortParams.direction]
+              ) : (
+                <span style={{ visibility: 'hidden' }}>
+                  {orderSymbols.ascending}
+                </span>
+              )}
```

Comment:
@LostInStatic just wanted to ask for a bit of clarification as to why there is the need to render the invisible `span` in such case.

Reply:
@LostInStatic if you want ot keep the centered things dead center you can maybe add another span on the other side of the element and that should do the trick.

Reply:
@karczewskiPiotr I went with a flexbox solution as discussed, the final effect IMO looks allright:


![electron_7X2jfNyfHH](https://user-images.githubusercontent.com/23729453/203417906-f1e00ac5-1ed2-4f0e-b976-8a92d607eba4.gif)

Reply:
@LostInStatic that looks good 👀